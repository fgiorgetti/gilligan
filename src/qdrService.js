/*
Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
*/

import RESTService from "./restService";
import Adapter from "./adapter";
import { utils } from "./utilities";
export const UPDATE_INTERVAL = 2000;
export const ALERT_TIMEOUT = 6000;

// number of milliseconds between topology updates
export class QDRService {
  constructor(hooks) {
    this.hooks = hooks;
    this.history = {};
    this.VAN = null;
    this.rest = null;
    this.adapter = null;
    this.siteInfo = null;
  }

  connect = (to) => {
    return new Promise((resolve, reject) => {
      if (!this.rest) this.rest = new RESTService();
      this.rest
        .getData(to)
        .then(
          (data) => {
            if (this.adapter) delete this.adapter;
            this.adapter = new Adapter(data);
            this.VAN = data;
            this.saveTimeSeries(data);
            this.initColors(data);
            if (!this.siteInfo) {
              this.rest.getSiteInfo().then((info) => {
                this.siteInfo = info;
                resolve(data);
              });
            } else {
              resolve(data);
            }
          },
          (error) => reject(error)
        )
        .catch(() => {});
    });
  };

  // SITE
  getSiteInfo = () => this.rest.getSiteInfo();

  // USETOKEN
  uploadToken = (data) => this.rest.uploadToken(data);

  getSkupperTokenURL = () => {
    return this.rest.getSkupperTokenURL();
  };

  // GENERATE_TOKEN
  getTokenData = () => this.rest.getTokenData();

  // DELETE_TOKEN
  deleteToken = (data) => this.rest.deleteToken(data);

  // UPDATE_TOKEN
  updateToken = (data) => this.rest.updateToken(data);

  // UNLINK
  unlinkSite = (data) => this.rest.unlinkSite(data);

  update() {
    return this.connect();
  }

  initColors = (data) => {
    data.sites.forEach((site) => {
      utils.siteColor(site.site_name, site.site_id);
    });
    data.services.forEach((service) => {
      utils.serviceColor(service.address);
    });
  };

  saveTimeSeries = (data) => {
    // push the current link.request onto link.history
    data.deploymentLinks.forEach((link) => {
      utils.keepHistory({
        obj: link.request,
        storage: this.history,
        key: link.key,
        history: (60 * 60 * 1000) / UPDATE_INTERVAL, // 1 hour of samples
      });
      link.history = this.history[link.key];
    });
  };
}

(function () {
  console.dump = function (o) {
    if (window.JSON && window.JSON.stringify)
      console.log(JSON.stringify(o, undefined, 2));
    else console.log(o);
  };
})();
