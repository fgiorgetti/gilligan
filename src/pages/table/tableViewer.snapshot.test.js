/*
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
*/
import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TableViewer from "./tableViewer";
import { QDRService } from "../../qdrService";

const handleShowSubTable = jest.fn();
const setOptions = jest.fn();
const service = new QDRService();
const props = {
  view: "service",
  mode: "table",
  service,
  setOptions,
  handleShowSubTable,
};

it("matches service table snapshot", () => {
  return service.connect().then(async () => {
    const { queryByTestId } = await render(<TableViewer {...props} />);
    const page = queryByTestId(/data-testid_service/i).outerHTML;
    expect(page).toMatchSnapshot();
  });
});

it("matches site table snapshot", () => {
  props.view = "site";
  return service.connect().then(async () => {
    const { queryByTestId } = await render(<TableViewer {...props} />);
    const page = queryByTestId(/data-testid_site/i).outerHTML;
    expect(page).toMatchSnapshot();
  });
});

it("matches deployment table snapshot", () => {
  props.view = "deployment";
  return service.connect().then(async () => {
    const { queryByTestId } = await render(<TableViewer {...props} />);
    const page = queryByTestId(/data-testid_deployment/i).outerHTML;
    expect(page).toMatchSnapshot();
  });
});
