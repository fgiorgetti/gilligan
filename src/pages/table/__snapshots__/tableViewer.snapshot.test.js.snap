// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches deployment table snapshot 1`] = `"<table data-testid=\\"data-testid_deployment\\" aria-label=\\"deployment\\" role=\\"grid\\" class=\\"pf-c-table pf-m-grid-md pf-m-compact\\" data-ouia-component-type=\\"PF4/Table\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Table-3\\"><thead class=\\"\\"><tr><th scope=\\"col\\" data-label=\\"Name (site)\\" data-key=\\"0\\" aria-sort=\\"ascending\\" class=\\"pf-c-table__sort pf-m-selected\\"><button type=\\"button\\" class=\\"pf-c-table__button\\"><div class=\\"pf-c-table__button-content\\"><span class=\\"pf-c-table__text\\">Name (site)</span><span class=\\"pf-c-table__sort-indicator\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 256 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z\\"></path></svg></span></div></button></th><th scope=\\"col\\" data-label=\\"Protocol\\" data-key=\\"1\\" aria-sort=\\"none\\" class=\\"pf-c-table__sort\\"><button type=\\"button\\" class=\\"pf-c-table__button\\"><div class=\\"pf-c-table__button-content\\"><span class=\\"pf-c-table__text\\">Protocol</span><span class=\\"pf-c-table__sort-indicator\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 256 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z\\"></path></svg></span></div></button></th><th scope=\\"col\\" data-label=\\"Site\\" data-key=\\"2\\" aria-sort=\\"none\\" class=\\"pf-c-table__sort\\"><button type=\\"button\\" class=\\"pf-c-table__button\\"><div class=\\"pf-c-table__button-content\\"><span class=\\"pf-c-table__text\\">Site</span><span class=\\"pf-c-table__sort-indicator\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 256 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z\\"></path></svg></span></div></button></th></tr></thead><tbody class=\\"\\"><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-14\\"><td data-label=\\"Name (site)\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"cartservice (site-b)\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-14\\">cartservice (site-b)</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td><td data-label=\\"Site\\" data-key=\\"2\\" class=\\"\\">site-b</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-15\\"><td data-label=\\"Name (site)\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"checkoutservice (site-b)\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-15\\">checkoutservice (site-b)</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td><td data-label=\\"Site\\" data-key=\\"2\\" class=\\"\\">site-b</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-16\\"><td data-label=\\"Name (site)\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"currencyservice (site-b)\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-16\\">currencyservice (site-b)</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td><td data-label=\\"Site\\" data-key=\\"2\\" class=\\"\\">site-b</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-17\\"><td data-label=\\"Name (site)\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"emailservice (mini1)\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-17\\">emailservice (mini1)</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td><td data-label=\\"Site\\" data-key=\\"2\\" class=\\"\\">mini1</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-18\\"><td data-label=\\"Name (site)\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"frontend (site-a)\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-18\\">frontend (site-a)</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td><td data-label=\\"Site\\" data-key=\\"2\\" class=\\"\\">site-a</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-19\\"><td data-label=\\"Name (site)\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"paymentservice (mini1)\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-19\\">paymentservice (mini1)</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td><td data-label=\\"Site\\" data-key=\\"2\\" class=\\"\\">mini1</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-20\\"><td data-label=\\"Name (site)\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"productcatalogservice (site-a)\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-20\\">productcatalogservice (site-a)</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td><td data-label=\\"Site\\" data-key=\\"2\\" class=\\"\\">site-a</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-21\\"><td data-label=\\"Name (site)\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"recommendationservice (site-a)\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-21\\">recommendationservice (site-a)</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td><td data-label=\\"Site\\" data-key=\\"2\\" class=\\"\\">site-a</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-22\\"><td data-label=\\"Name (site)\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"redis-cart (site-b)\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-22\\">redis-cart (site-b)</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">TCP</td><td data-label=\\"Site\\" data-key=\\"2\\" class=\\"\\">site-b</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-23\\"><td data-label=\\"Name (site)\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"shippingservice (mini1)\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-23\\">shippingservice (mini1)</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td><td data-label=\\"Site\\" data-key=\\"2\\" class=\\"\\">mini1</td></tr></tbody></table>"`;

exports[`matches service table snapshot 1`] = `"<table data-testid=\\"data-testid_service\\" aria-label=\\"service\\" role=\\"grid\\" class=\\"pf-c-table pf-m-grid-md pf-m-compact\\" data-ouia-component-type=\\"PF4/Table\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Table-1\\"><thead class=\\"\\"><tr><th scope=\\"col\\" data-label=\\"Name\\" data-key=\\"0\\" aria-sort=\\"ascending\\" class=\\"pf-c-table__sort pf-m-selected\\"><button type=\\"button\\" class=\\"pf-c-table__button\\"><div class=\\"pf-c-table__button-content\\"><span class=\\"pf-c-table__text\\">Name</span><span class=\\"pf-c-table__sort-indicator\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 256 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z\\"></path></svg></span></div></button></th><th scope=\\"col\\" data-label=\\"Protocol\\" data-key=\\"1\\" aria-sort=\\"none\\" class=\\"pf-c-table__sort\\"><button type=\\"button\\" class=\\"pf-c-table__button\\"><div class=\\"pf-c-table__button-content\\"><span class=\\"pf-c-table__text\\">Protocol</span><span class=\\"pf-c-table__sort-indicator\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 256 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z\\"></path></svg></span></div></button></th></tr></thead><tbody class=\\"\\"><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-1\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"cartservice\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-1\\">cartservice</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-2\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"checkoutservice\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-2\\">checkoutservice</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-3\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"currencyservice\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-3\\">currencyservice</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-4\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"emailservice\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-4\\">emailservice</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-5\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"frontend\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-5\\">frontend</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-6\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"paymentservice\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-6\\">paymentservice</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-7\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"productcatalogservice\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-7\\">productcatalogservice</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-8\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"recommendationservice\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-8\\">recommendationservice</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-9\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"redis-cart\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-9\\">redis-cart</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">TCP</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-10\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"shippingservice\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-10\\">shippingservice</button></td><td data-label=\\"Protocol\\" data-key=\\"1\\" class=\\"\\">HTTP</td></tr></tbody></table>"`;

exports[`matches site table snapshot 1`] = `"<table data-testid=\\"data-testid_site\\" aria-label=\\"site\\" role=\\"grid\\" class=\\"pf-c-table pf-m-grid-md pf-m-compact\\" data-ouia-component-type=\\"PF4/Table\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Table-2\\"><thead class=\\"\\"><tr><th scope=\\"col\\" data-label=\\"Name\\" data-key=\\"0\\" aria-sort=\\"ascending\\" class=\\"pf-c-table__sort pf-m-selected\\"><button type=\\"button\\" class=\\"pf-c-table__button\\"><div class=\\"pf-c-table__button-content\\"><span class=\\"pf-c-table__text\\">Name</span><span class=\\"pf-c-table__sort-indicator\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 256 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z\\"></path></svg></span></div></button></th><th scope=\\"col\\" data-label=\\"Namespace\\" data-key=\\"1\\" aria-sort=\\"none\\" class=\\"pf-c-table__sort\\"><button type=\\"button\\" class=\\"pf-c-table__button\\"><div class=\\"pf-c-table__button-content\\"><span class=\\"pf-c-table__text\\">Namespace</span><span class=\\"pf-c-table__sort-indicator\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 256 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z\\"></path></svg></span></div></button></th><th scope=\\"col\\" data-label=\\"Edge\\" data-key=\\"2\\" aria-sort=\\"none\\" class=\\"pf-c-table__sort\\"><button type=\\"button\\" class=\\"pf-c-table__button\\"><div class=\\"pf-c-table__button-content\\"><span class=\\"pf-c-table__text\\">Edge</span><span class=\\"pf-c-table__sort-indicator\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 256 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M214.059 377.941H168V134.059h46.059c21.382 0 32.09-25.851 16.971-40.971L144.971 7.029c-9.373-9.373-24.568-9.373-33.941 0L24.971 93.088c-15.119 15.119-4.411 40.971 16.971 40.971H88v243.882H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.568 9.373 33.941 0l86.059-86.059c15.12-15.119 4.412-40.971-16.97-40.971z\\"></path></svg></span></div></button></th></tr></thead><tbody class=\\"\\"><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-11\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"mini1\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-11\\">mini1</button></td><td data-label=\\"Namespace\\" data-key=\\"1\\" class=\\"\\">default</td><td data-label=\\"Edge\\" data-key=\\"2\\" class=\\"\\">false</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-12\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"site-a\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-12\\">site-a</button></td><td data-label=\\"Namespace\\" data-key=\\"1\\" class=\\"\\">pub1</td><td data-label=\\"Edge\\" data-key=\\"2\\" class=\\"\\">false</td></tr><tr class=\\"\\" data-ouia-component-type=\\"PF4/TableRow\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-TableRow-13\\"><td data-label=\\"Name\\" data-key=\\"0\\" class=\\"\\"><button data-testid=\\"site-b\\" aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-primary link-button\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-primary-13\\">site-b</button></td><td data-label=\\"Namespace\\" data-key=\\"1\\" class=\\"\\">pub2</td><td data-label=\\"Edge\\" data-key=\\"2\\" class=\\"\\">false</td></tr></tbody></table>"`;
